pywinauto/controls/uia_controls.py

```diff
@@ -1212,10 +1216,7 @@ def texts(self):
     def button_count(self):
         """Return a number of buttons on the ToolBar"""
         if self.win32_wrapper is not None:
-            btn_count = self.win32_wrapper.button_count()
-            if btn_count:
-                return btn_count
-            return len(self.win32_wrapper.children())
+            return len(self.buttons())
```

Comment:
This is less optimal, because `self.buttons()` walks all buttons and creates a list, but we need only count!

Reply:
This is kind of a workaround because `win32_wrapper.button_count()` takes into account duplicate buttons

Reply:
Ah, OK. Then please add a comment about it.